src/main/java/de/bonndan/nivio/output/map/hex/GroupAreaFactory.java

```diff
@@ -34,37 +35,57 @@
 
         Set<Item> items = group.getItems();
         Set<Hex> inArea = new HashSet<>();
+        List<Item> connected = new ArrayList<>();
 
+        if (!items.iterator().hasNext()) {
+            LOGGER.warn("Could not determine group area for group {}", group);
+            return inArea;
+        }
         //surround each item
-        items.forEach(item -> {
-            Hex hex = allVertexHexes.get(item);
+        Item next = items.iterator().next();
+        while (next != null) {
+
+            LOGGER.debug("adding {} to group area", next);
+            Hex hex = allVertexHexes.get(next);
             inArea.add(hex);
             hex.neighbours().forEach(neigh -> {
                 if (!occupied.contains(neigh))
                     inArea.add(neigh);
             });
 
-            Set<Hex> closestNeighbours = getClosestItemsHexes(item, items, allVertexHexes);
+            Optional<Item> closest = getClosestItem(next, items, allVertexHexes, connected);
+            if (closest.isEmpty()) {
+                LOGGER.debug("no closest item found for {}", next);
+                break;
+            }
+
             // we dont care for occupied tiles here, since we just want the closest item within group, and non-group
             // items cannot be anywhere nearby (other types of obstacles do not exist yet)
             PathFinder pathFinder = new PathFinder(Set.of());
-            closestNeighbours.forEach(neighbour -> {
-                Optional<HexPath> path = pathFinder.getPath(hex, neighbour);
-                if (path.isEmpty()) {
-                    return;
-                }
+
+            Hex destination = allVertexHexes.get(closest.get());
+            Optional<HexPath> path = pathFinder.getPath(hex, destination);
+            if (path.isPresent()) {
                 Set<Hex> padded = new HashSet<>(); //pad to avoid thin bridges, also workaround for svh outline issue
                 path.get().getHexes().forEach(pathTile -> {
                     padded.add(pathTile);
                     padded.addAll(pathTile.neighbours());
                 });
                 padded.stream().filter(hex1 -> !occupied.contains(hex1)).forEach(inArea::add);
-            });
+            }
 
-        });
+            connected.add(next);
+            // stop if the next one has been connected already
+            next = connected.contains(closest.get()) ? null : closest.get();
+        }
+
+        // adding hexes with many sides adjacent to group area until no more can be added
 
-        Set<Hex> bridges = getBridges(inArea);
-        inArea.addAll(bridges);
+        Set<Hex> bridges = getBridges(inArea, 2);
+        while (!bridges.isEmpty()) {
+            inArea.addAll(bridges);
+            bridges = getBridges(inArea, 3); // 2 might be too aggressive and collide with other group areas
+        }
 
         return inArea;
```

Comment:
This logic is hard to review. Maybe it is best if we talk this through in a call? 

Are our tests good enough? I saw that you did not need to make many changes there :)